### PR TITLE
add test case

### PR DESCRIPTION
**Pull request by LeungChiNan**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation and Context

add test case

## Description

add test case

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: diopi_test/python/conformance/conformance_test.py [hunk 1]
```diff
@@ -164,13 +167,13 @@ def test_dropout_(func, input, p=0.5, training=True, inplace=False):
                 mask_numpy = mask_numpy * tmp
             remains = out_numpy[mask_numpy == 1]
             ref = input_numpy[mask_numpy == 1]
-            assert np.allclose(remains, ref / (1 - p), rtol=1e-4, atol=1e-5),\
+            assert np.allclose(remains, ref / (1 - p), rtol=rtol, atol=atol),\
                 f"failed to execute {name}"
-
-            assert np.abs(real_ratio - (1 - p)) < 3e-2,\
-                f"failed to execute {name} "
+            if mask.numel() > 100:
```
Comment (NeosZhang) [inline diopi_test/python/conformance/conformance_test.py:172]: 这个100和后续的3e-2不写成硬编码，用变量表示更好一些

Comment (LeungChiNan) [reply diopi_test/python/conformance/conformance_test.py:172]: 写成变量反而不直观

File: diopi_test/python/conformance/conformance_test.py [hunk 2]
```diff
@@ -155,7 +155,10 @@ def test_dropout_(func, input, p=0.5, training=True, inplace=False):
         out_numpy = out.numpy()
         mask_numpy = mask.numpy()
 
-        if training:
+        rtol = 1e-2 if input_numpy.dtype == np.float16 else 1e-4
```
[the remark this answers is not in the document]
Comment (LeungChiNan) [reply diopi_test/python/conformance/conformance_test.py]: 改用了utils里的配置

File: diopi_test/python/conformance/conformance_test.py [hunk 3]
```diff
@@ -164,13 +167,13 @@ def test_dropout_(func, input, p=0.5, training=True, inplace=False):
                 mask_numpy = mask_numpy * tmp
             remains = out_numpy[mask_numpy == 1]
             ref = input_numpy[mask_numpy == 1]
-            assert np.allclose(remains, ref / (1 - p), rtol=1e-4, atol=1e-5),\
+            assert np.allclose(remains, ref / (1 - p), rtol=rtol, atol=atol),\
```
[the remark this answers is not in the document]
Comment (LeungChiNan) [reply diopi_test/python/conformance/conformance_test.py:170]: 改用了utils里的配置

**Review by yangbofun — APPROVED** [review]

**Pull request merged**